docs/genesis-preparation.md

```diff
@@ -0,0 +1,352 @@
+# Genesis Preparation
+
+This article describes the steps should be taken to prepare genesis block.
+
+## 1. Nodes Keys
+
+First of all, secret/public keys for the cluster's nodes should be generated. Nodes will
+use them to perform blockchain-related procedures (for examples, to sign new blocks).
+These keys should be uploaded to the nodes during cluster deployment.
+
+### 1.1. Secret Keys
+
+To generate secret keys for nodes use this command:
+
+```
+$ stack exec -- cardano-keygen generate-key --path PATH
+```
+
+where `PATH` is a path to file for generated key. Use `.key` extension for it.
+
+For exmaple, to generate keys for 7 nodes execute this command 7 times:
+
+```
+$ stack exec -- cardano-keygen generate-key --path /tmp/node1.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node2.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node3.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node4.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node5.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node6.key
+$ stack exec -- cardano-keygen generate-key --path /tmp/node7.key
+```
+
+**Responsible**: The secret key for particular node should generate a person who
+launch this node. For Mainnet launching secret keys generation should be done by
+IOHK DevOps team.
+
+### 1.2. Public Keys
+
+It's possible to read generated secret key by this command:
+
+```
+$ stack exec -- cardano-keygen read-key --path /tmp/node1.key
```

Comment:
In `avieth/csl1617_config` this also requires `--system-start` argument.

That's probably a bug, but if it's not going to be fixed before the dry-run it should be included in the docs.

Reply:
It's a bug, but a harmless one

Reply:
`--system-start 0`